patternlab/styleguide/source/_patterns/01-atoms/09-media/figure.twig

```diff
@@ -1,16 +1,63 @@
 {% set figureClass = figure.align ? "ma__figure--" ~ figure.align : "" %}
+{% set iframeClass = "ma__iframe__container js-ma-responsive-iframe" %}
+
+{# Dataviz embed iframe #}
+{# iframe element is generaged by Tableau JS. #}
+{% if figure.dataviz %}
+  {# Set align default to left. #}
+  {% set figureClass = figure.align ? "ma__figure--" ~ figure.align : "ma__figure--left" %}
+  {% set id = figure.dataviz.id %}
+  {% set src = figure.dataviz.src %}
+  {# TODO:  SET SIZE VALUES #}
+  {% set size = figure.dataviz.size ? "ma__figure--" ~ figure.dataviz.size : "ma__figure--large" %}
+{% endif %}
+
+{# <figure class="ma__figure {{ figureClass }}"> #}
+{# {% if figure.dataviz %}<div class="ma__iframe--responsive">{% endif %} #}
+<figure class="ma__figure {{ figureClass }}{% if figure.iframe %} {{ iframeClass }}{% endif %}{% if figure.dataviz %} {{ size }} ma__dataviz{% endif %}"{% if figure.dataviz %} id="{{ id }}"{% endif %}>
+{# <figure class="ma__figure {{ figureClass }}{% if figure.iframe %} {{ iframeClass }}{% endif %}{% if figure.dataviz %} {{ size }} ma__dataviz js-ma-responsive-iframe{% endif %}"{% if figure.dataviz %} id="{{ id }}"{% endif %}> #}
+
 
-<figure class="ma__figure {{ figureClass }}">
-  {% set image = figure.image %}
   {% if figure.title %}
   <span class="ma__figure__title">
     {{ figure.title }}
   </span>
   {% endif %}
-  {% include "@atoms/09-media/image.twig" %}
+
+  {# For image. #}
+  {% if figure.image %}

```

Comment:
Ooh... This is a lot of conditional logic in what should be a very basic "fundamental" template.  I know we've got this kind of pattern all over Mayflower, but in my opinion, `figure` should not have any opinion about what goes inside of it.  Instead, it should allow you to inject whatever kind of thing you want, and allow the class to be set accordingly.  Eg:
```twig
# figure.twig
{% set alignmentClass = figure.align ? "ma__figure--" ~ figure.align : "" %}

<figure class="ma__figure {{ alignmentClass }}{{ figure.class ? figure.class ~" " : "" }}">
  {% if figure.title %}
  <span class="ma__figure__title">
    {{ figure.title }}
  </span>
  {% endif %}
  {{ figure.content }}
  {% if figure.caption %}
    <figcaption class="ma__figure__caption">
      {{ figure.caption }}
    </figcaption>
  {% endif %}
</figure>
```
Then, we could create wrapper components for the different types of figures:
```twig
{# figure--image.twig #}
{% if figure.image %}
  {# Render figure.content, which will be the main content of the figure #}
  {# Important: we wrap this in an `if` statement to allow consuming systems
     to either use figure.image or figure.content (if they prefer to externally
     render figure.content
   #}
  {% set figure = figure|merge({
    content: include("@atoms/09-media/image.twig", {image: figure.image})
  }) %}
{% endif %}
{% include "figure.twig" with {
  figure: figure|merge({
    class: "ma__iframe__container js-ma-responsive-iframe"
  })
} %}

{# figure--iframe.twig #}
{% if figure.iframe %}
  {# Render figure.content, which will be the main content of the figure #}
  {# Important: we wrap this in an `if` statement to allow consuming systems
     to either use figure.iframe or figure.content (if they prefer to externally
     render figure.content
   #}
  {% set figure = figure|merge({
    content: include("@atoms/09-media/iframe.twig", {iframe: figure.iframe}),
  }) %}
{% endif %}
{% include "figure.twig" %}

{# figure--dataviz.twig #}
{% if figure.dataviz %}
  {# Render figure.content, which will be the main content of the figure #}
  {# Important: we wrap this in an `if` statement to allow consuming systems
     to either use figure.iframe or figure.content (if they prefer to externally
     render figure.content
   #}
  {% set content %}
    <script type="text/javascript">
      window.addEventListener("DOMContentLoaded", function() {
        new tableau.Viz(document.getElementById("{{ figure.dataviz.id }}"), "{{ figure.dataviz.url }}");
      });
    </script>
  {% endset %}
  {% set figure = figure|merge({
    content: content,
  }) %}
{% endif %}
{% include "figure.twig" with {
  figure: figure|merge({
    class: "ma__iframe__container js-ma-responsive-iframe"
  })
} %}
```
I know this is a pretty big deviation from how we've done things in other parts of Mayflower, but this would allow us to reuse a very simple core "figure" template, while still allowing for complex variation through the use of these wrapper templates.  Additionally, it allows us to write the Tableau embed code in Drupal, then inject it into `figure--dataviz.twig` via the `content` variable, which is what we really want to do.  

What do you think?